homeassistant/components/stream/__init__.py

```diff
@@ -140,7 +140,7 @@ def hls_output(self) -> StreamOutput:
     def _hls_idle(self):
         """Reset access token and cleanup stream due to inactivity."""
         self.access_token = None
-        if not self.keepalive:
+        if not self.keepalive and self._hls:
             self._hls.cleanup()
             self._hls = None
             self._hls_timer = None
```

Comment:
Do we need to clear `self._hls_timer` here?

Reply:
Clear should only be required if the alarm did not previously fire which sets `_unsub` to None.  However, it does not hurt since its a no-op.

It is not clear to me if it improves readability to leave it out, or if it is better to be defensive just in case. I left it out, but I could go either way.  I've added a comment to `clear` to indicate that it only does something when the alarm has not fired yet.

Reply:
Sorry I didn't read it clearly, this is the callback